### PR TITLE
[SYCL] Improve checks of kernel execution range in INT_MAX limit.

### PR DESCRIPTION
**Pull request by s-kanaev**
Improved check is for the following use-cases. When we have a lot of small work-groups the number of work-items may still be out of INT_MAX limit Also, sum of range and offset may exceed the limitation while each value is within INT_MAX limit on its own.

### CODE REVIEW

File: clang/test/Preprocessor/sycl-macro.cpp
Comment (elizabethandrews) [inline clang/test/Preprocessor/sycl-macro.cpp:10]: I don't think the run lines 2-6 are needed now that it is on by default.

**Review by elizabethandrews — COMMENTED** [review]
Other than comment on test, FE changes LGTM

File: clang/test/Preprocessor/sycl-macro.cpp
Comment (s-kanaev) [reply clang/test/Preprocessor/sycl-macro.cpp:10]: Removed.

**Comment by s-kanaev** [comment]
@premanandrao @v-klochkov @Fznamznon @intel/llvm-reviewers-runtime  ping

**Review by Fznamznon — DISMISSED** [review]

**Review by alexbatashev — APPROVED** [review]

**Comment by s-kanaev** [comment]
The turn on of the assumptions was excluded from this patch.

**Comment by Fznamznon** [comment]
Please do not wait for review from me.

**Review by alexbatashev — APPROVED** [review]

**Pull request merged**
